copylot/hardware/cameras/basler/camera.py

```diff
@@ -0,0 +1,557 @@
+# -*- coding: utf-8 -*-
+"""
+
+@author: yang liu
+
+Purpose: BaslerCamera is a wrapper for Basler pypylon, inheriting from PyCamera
+to standardize a simple API for integration of cameras into instruments run by a
+python framework. part of the code is borrowed from
+https://github.com/czbiohub-sf/pyCameras/blob/master/py_cameras/py_basler.py
+
+"""
+
+from copylot.hardware.cameras.abstract_camera import AbstractCamera
+from pypylon import pylon
+from pypylon import genicam
+import sys
+from copylot import logger
+import numpy as np
+from enum import Enum
+from time import perf_counter, sleep
+
+
+exitcode = 0
+
+# ---------------------------------------CONSTANTS-----------------------------------------------
+_CAM_TRIG_SLEEP = 0.01
+_CAM_TIMEOUT_MS = 1000
+_DEF_START_EXPOSURE_MS = 10
+_MAX_CAMLIST_ATTEMPTS = 5
```

Comment:
All of these should be class members, do you have a very strong reason to have these constants global @YangLiujames ?

Reply:
I can remove the global settings, these are just from the code that I borrowed from Emily